src/compile.js

```diff
@@ -1809,7 +1810,7 @@ Compiler.prototype.cclass = function (s) {
     entryBlock = this.newBlock("class entry");
 
     this.u.prefixCode = "var " + scopename + "=(function $" + s.name.v + "$class_outer($globals,$locals,$rest){var $gbl=$globals,$loc=$locals;";
-    this.u.switchCode += "return(function " + s.name.v + "(){";
+    this.u.switchCode += "return(function " + fixReservedNames(s.name.v) + "(){";
```

Comment:
The more I think about this, maybe the easiest/safest solution is to not name the function at all.  As best as I can tell, the name isn't necessary.  If we want a name for stack-trace/debugging purposes, maybe we name it something descriptive yet "safe" like `"$" + s.name.v + "$_closure"`